### ISSUE TITLE
[Bug] Ktuvit Add-on stopped working since last Stremio update

### ISSUE THREAD

**Issue by TwilightMercy** [maormagori/ktuvit-stremio#42]
היי מאור,

מעדכן אותך בבעיה שחווינו מאז העדכון האחרון שיצא ל-Stremio השבוע: 
https://blog.stremio.com/stremio-tech-update-21-stremio-web-release-android-tv-update-stremio-on-iphone-and-more/

מאז לא מקבלים כתוביות מKtuvit עבור כל המקורות,
ב-Wizdom הכל עובד תקין. 

תודה לך!

**Comment by maormagori** [comment]
It shouldn't be related to the new Stremio version since there was no change to their addon protocol.
It appears that the endpoint used to search content on Ktuvit has stopped working:
```
https://www.ktuvit.me/Services/ContentProvider.svc/SearchPage_search
```

This might be an issue worth discussing with Ktuvit.
I'll try to investigate more and update here. Hopefully, this isn't a sign of a more significant issue.

**Comment by TwilightMercy** [comment]
It is weird,
In Kodi the subtitles addon uses the same URL, and everything is working fine.

**Comment by maormagori** [comment]
I can update that the Ktuvit endpoint is working as expected. I had problems with my testing.
@TwilightMercy Can you confirm that this is only happening in the new version?
If so, what platform have you tested (Android, Android TV, PC...)?

It would be great if you could provide a specific movie/series that you see this happening on.
I'll continue testing and update if I find something.

**Comment by TwilightMercy** [comment]
It doesn’t seem to be related to new Stremio version. An Android TV device with pre-updated Stremio has been tested. 

The addon stopped working for all content and tested on 3 platforms: 
Windows PC, Android Phone, Android TV

All at once stopped getting results from Ktuvit, while Wizdom is working. 

if you want specific example:
John Wick 4 (2023)

![image](https://github.com/maormagori/ktuvit-stremio/assets/29895680/69429e60-dc29-4382-9ec1-8e6076db0d98)

**Comment by maormagori** [comment]
Yeah ok found the bug and this one's on me 🤦
The new client dropped the extra information about the file being watched by the user, aka the `fileName` attribute.
I was using this attribute to sort the subtitles by most to least relevant.
And in[ this line](https://github.com/maormagori/ktuvit-stremio/blob/0ffd2d1210fac0a0ce71155b40d9dce0fc4494ff/routes/subs.js#L126) I'm supposed to return the unsorted array if there's no `fileName` attribute (As I stated I'd do in the log line preceding it).
I accidentally returned nothing instead.

This is an easy fix but it implies a bigger change. 
### As of right now, neither this addon nor the wisdom one can implement any sorting to the subtitles.

I've asked the Stremio devs for some clarity regarding the change.
I'll update you guys here if they answer.

**Comment by TwilightMercy** [comment]
I managed to read the comment before you deleted it, so it does related to new Stremio version?

Something has changed regarding to fileName attribute from Stremio?

**Comment by maormagori** [comment]
Yeah decided to investigate further before doing changes. Will update you guys 👍

**Comment by TwilightMercy** [comment]
Thanks. 👍🏻🙏🏻

**Comment by maormagori** [comment]
### Addon should be back up. Please check and update me.

To those interested, here are my findings (For those **really** interested):
- I was unable to reproduce the addon's behavior locally, meaning something was up in the production environment.
Due to not implementing #19 I can't really tell what's happening in the prod environment. But, I have a suspicion. One of the main differences when running in production mode is there is [more extensive logging saved to a local file](https://github.com/maormagori/ktuvit-stremio/blob/7df6cb15b804e394c5e5b28daf7304619971b16f/common/logger.js#L24).
The file probably was being piled up until reaching a point where logging into it threw a maxfile error so the addon was crashing on every request. I was more assured when the addon started working normally when I redeployed it, so I disabled local logging in #44 and #45.
- Subtitle sorting was disabled in the addon. Probably forgot to enable it in the past. It's enabled now BUT:
- The new client doesn't send the `fileName` parameter. I noticed this started to happen when the new Android TV client was published and it looks like the new web client doesn't send it as well. This effectively disables the addons ability to sort subtitles. I've contacted Stremio devs asking why they changed this behavior, but I don't think we'll get a satisfying response here. Either way, I'll update here if I get an answer from them.

The bottom line, we should be back to business. Thanks, @TwilightMercy for notifying me 👌

**Comment by TwilightMercy** [comment]
Ktuvit subtitles are back 👌🏻

![image](https://github.com/maormagori/ktuvit-stremio/assets/29895680/15ef4c42-18fa-45aa-ac7f-4b38ad3e1c7a)

**Comment by TwilightMercy** [comment]
So, up until now the sorting wasn’t enabled in Ktuvit addon? (And also Wizdom?)

And now that it is enabled, based on what parameter the subtitles are sorted by? If no fileName parameter is being sent to the addon?

**Comment by maormagori** [comment]
Thanks for verifying!
The sorting was only disabled on this addon. I enabled it in the last deployment.
Requests made by previous clients would receive sorted subtitles, only the new clients are dropping the `fileName` parameter so for them, the returned subtitles won't be sorted.

I agree that this is upsetting but I can only work with the info I'm given and Stremio decided to omit this piece of info🫤

**Comment by TwilightMercy** [comment]
That’s really upsetting... Just to confirm, it affects sorting for both Ktuvit and Wizdom right?

And if that’s ok to ask, did you contact Stremio devs about it in GitHub issue / other method? If so, I would like to follow their response too.

**Comment by maormagori** [comment]
Yeah, both of the addons rely on this argument to sort the subtitles.

I sent a message in the #devs channel of their [discord](https://discord.gg/zNRf6YF), not sure how active it is these days but they used to be pretty responsive there.

**Comment by TwilightMercy** [comment]
Great, thank you. 

From their discord channel - I found links to these two active issues pages:

https://github.com/Stremio/stremio-bugs/issues

https://github.com/Stremio/stremio-features/issues

If that helps.

**Comment by maormagori** [comment]
Great suggestion, I opened an issue on the stremio-addons-sdk repo as it's the most relevant to the problem:
https://github.com/Stremio/stremio-addon-sdk/issues/221
We can follow progress there.

I'm closing this issue for now. Feel free to contact me if you still have any more questions.

**Comment by TwilightMercy** [comment]
Thank you very much. 

And in the meantime..

![image](https://github.com/maormagori/ktuvit-stremio/assets/29895680/21482c96-ebbb-4242-ad04-e03eb3140520)

**Comment by TwilightMercy** [comment]
Hi friend,

Seems that are no results from Ktuvit addon again (Wizdom works) :)

And in your issue regarding Stremio update changes - I see that it might take some time..

**Comment by maormagori** [comment]
Thanks for informing me @TwilightMercy !
I've opened #46 and will update it there to keep it organized.

**Comment by TwilightMercy** [comment]
Thank you! I’ll be following the issue for updates 🙏🏼
